src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -479,6 +482,18 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
       if (__DEV__) {
+        const Component = workInProgress.type;
+
+        if (Component) {
+          warning(
+            !Component.childContextTypes,
+            '%s(...): childContextTypes cannot be defined on a functional component.',
+            Component.displayName || Component.name || 'Component'
+          );
+          if (Component.childContextTypes) {
+            warnAboutMissingGetChildContext(workInProgress);
```

Comment:
Agreed, completely. I was unsure if we wanted 1:1 parity with Stack in regards to these strange warning messages. I don't know why anyone would ever need the second warning message when using functional components, given they can't ever use `getChildContext` on the components. Was there intentions at some point to allow functional components to use `getChildContext`?